dio/lib/src/dio.dart

```diff
@@ -3,6 +3,7 @@ import 'dart:async';
 import 'adapter.dart';
 import 'cancel_token.dart';
 import 'dio/dio_for_native.dart'
+    if (dart.library.js_interop) 'dio/dio_for_browser.dart'
```

Comment:
@AlexV525 I think we need to remove this import line everywhere. It now imports the old version specifically for WASM. I don't fully understand, why it was compiling with WASM before this was added.
After we actually add WASM support, we can add it back.

Reply:
If this is the reason, it could be additional compile checks are applied when the library is actually getting imported.